src/main/java/org/traccar/forward/EventForwarderMQTT.java

```diff
@@ -0,0 +1,108 @@
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.UUID;
+
+public class EventForwarderMQTT implements EventForwarder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventForwarderMQTT.class);
+    private final Mqtt5AsyncClient client;
+    private final ObjectMapper objectMapper;
+
+    private final String topic;
+
+    public EventForwarderMQTT(Config config, ObjectMapper objectMapper) {
+        Mqtt5SimpleAuth simpleAuth = null;
+        URI url;
+        try {
+            url = new URI(config.getString(Keys.EVENT_FORWARD_URL));
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+
+        String userInfo = url.getUserInfo();
+
+        if (userInfo != null) {
+            int delimiter = userInfo.indexOf(':');
+            if (delimiter == -1) {
+                LOGGER.error("Wrong credentials. Should be in format \"username:password\"");
+            } else {
+                simpleAuth = Mqtt5SimpleAuth.builder()
+                    .username(userInfo.substring(0, delimiter++))
+                    .password(userInfo.substring(delimiter).getBytes())
+                    .build();
+            }
+        }
+
+        String host = url.getHost();
+        int port = url.getPort();
+        client = Mqtt5Client.builder()
+                .identifier(UUID.randomUUID().toString())
+                .serverHost(host)
+                .serverPort(port)
+                .simpleAuth(simpleAuth)
+                .buildAsync();
+
+        this.objectMapper = objectMapper;
+        topic = config.getString(Keys.EVENT_FORWARD_TOPIC);
+        LOGGER.info("MQTT EventForwarder created with URL: " + host + ":" + port);
+    }
+
+    @Override
+    public void forward(EventData eventData, ResultHandler resultHandler) {
+        try {
+            byte[] payload = objectMapper.writeValueAsString(eventData).getBytes();
+            client.connect()
+                    .whenComplete((msg, ex) -> {
+                        if (ex != null) {
+                            LOGGER.error(ex.getMessage());
+                        } else {
+                            LOGGER.debug("Connected");
+                        }
+                    })
+                    .thenCompose(connAck -> client.publishWith()
+                            .topic(topic)
+                            .qos(MqttQos.AT_LEAST_ONCE)
+                            .payload(payload)
+                            .send()
+                            .whenComplete((msg, ex) -> {
+                                if (ex != null) {
+                                    LOGGER.error(ex.getMessage());
+                                } else {
+                                    LOGGER.debug("Send message");
+                                }
+                            }))
+                    .thenCompose(publishResult -> client.disconnect());
+        } catch (JsonProcessingException e) {
```

Comment:
This catch can probably be just around the JSON mapping.

Reply:
ok